build/azure-pipelines/darwin/sql-product-build-darwin.yml

```diff
@@ -0,0 +1,142 @@
+      archiveFile: '$(Build.BinariesDirectory)/source.tar.gz'
+
+  - task: PublishBuildArtifacts@1 # WHY ARE WE DOING THIS?
+    displayName: 'Publish Artifact: build scripts source'
+    inputs:
+      PathtoPublish: '$(Build.BinariesDirectory)/source.tar.gz'
+      ArtifactName: source
+
+  - script: |
+      set -e
+      cd $(Build.SourcesDirectory)/../azuredatastudio-darwin
+      zip -r -y $(Build.ArtifactStagingDirectory)/azuredatastudio-darwin.zip "Azure Data Studio.app"
+    displayName: 'Stable - Run zip'
+    condition: and(succeeded(), eq(variables['VSCODE_QUALITY'], 'stable'))
+
+  - script: |
+      set -e
+      cd $(Build.SourcesDirectory)/../azuredatastudio-darwin
+      zip -r -y $(Build.ArtifactStagingDirectory)/azuredatastudio-darwin.zip "Azure Data Studio - Insiders.app"
+    displayName: 'Insiders - Run zip'
+    condition: and(succeeded(), eq(variables['VSCODE_QUALITY'], 'insider'))
+
+  - task: SFP.build-tasks.custom-build-task-1.EsrpCodeSigning@1
+    displayName: 'ESRP CodeSigning'
+    inputs:
+      ConnectedServiceName: 'Code Signing'
+      FolderPath: '$(Build.ArtifactStagingDirectory)'
+      Pattern: 'azuredatastudio-darwin.zip'
+      signConfigType: inlineSignParams
+      inlineOperation: |
+        [
+          {
+            "keyCode": "CP-401337-Apple",
+            "operationSetCode": "MacAppDeveloperSign",
+            "parameters": [],
+            "toolName": "sign",
+            "toolVersion": "1.0"
+          }
+        ]
+      SessionTimeout: 20
+
+  - script: | # WHY ARE WE DOING THIS?
+      set -e
+      zip -d $(Build.ArtifactStagingDirectory)/azuredatastudio-darwin.zip "Azure Data Studio.pkg"
+    displayName: 'Stable - Remove .pkg from signed zip'
+    condition: and(succeeded(), eq(variables['VSCODE_QUALITY'], 'stable'))
+
+  - script: | # WHY ARE WE DOING THIS?
+      set -e
+      zip -d $(Build.ArtifactStagingDirectory)/azuredatastudio-darwin.zip "Azure Data Studio - Insiders.pkg"
+    displayName: 'Insiders - Remove .pkg from signed zip'
+    condition: and(succeeded(), eq(variables['VSCODE_QUALITY'], 'insider'))
+
+  - script: |
+      set -e
+      ./scripts/test.sh --build --coverage --reporter mocha-junit-reporter
+    displayName: Run unit tests
+    condition: and(succeeded(), eq(variables['RUN_TESTS'], 'true'))
+    timeoutInMinutes: 20
+    continueOnError: true
+
+  - script: | # WHY ARE WE DOING THIS?
+      set -e
+      BUILD="$(Build.SourcesDirectory)/../azuredatastudio-darwin"
+      PACKAGEJSON=`ls $BUILD/*.app/Contents/Resources/app/package.json`
+      VERSION=`node -p "require(\"$PACKAGEJSON\").version"`
+      COMMIT_ID=`git rev-parse HEAD`
+
+      echo -e "{  \"version\": \"$VERSION\", \"quality\": \"$VSCODE_QUALITY\", \"commit\": \"$COMMIT_ID\" }" > "$(Build.ArtifactStagingDirectory)/version.json"
+    displayName: 'Create version.json'
+
+  - script: | # WHY ARE WE DOING THIS?
+      set -e
+      for f in *
+      do
+      shasum -a 256 "$f" >> sha256hashes.txt
```

Comment:
These ones I'm less sure about. Maybe Karl would have some context on why these were added?

Reply:
We discussed this task in one of our sync-ups.  I don't have context on this, other than we should be publishing SHA hashed for the Linux RPM, DEB, and TAR.GZ files since that is how users verify downloads as Linux doesn't have a standard binary signing convention.  It doesn't appear this task is directly accomplishing this goal, but as Anthony mentioned, I'd prefer to do this migration in different steps.  (1) move over the tasks basically as-is then (2) clean-up things we no longer need.